File: lib-python/agi.py
__version__ = "$Revision$ $Date$"
__license__ = """
    Copyright (C) 2007, Proformatique

    This library is free software; you can redistribute it and/or
    modify it under the terms of the GNU Lesser General Public
    License as published by the Free Software Foundation; either
    version 2.1 of the License, or (at your option) any later version.

    This library is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
    Lesser General Public License for more details.

    You should have received a copy of the GNU Lesser General Public
    License along with this library; if not, write to the Free Software
    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
"""

def agi_escape_string(s):
	return s.replace('\\', '\\\\').replace('"', '\\"').replace('\n', ' ')

def agi_verbose(txt):
	print "VERBOSE \"%s\"" % agi_escape_string(txt)

def agi_set_variable(var, val):
        print "SET VARIABLE %s \"%s\"" %(agi_escape_string(var), agi_escape_string(val))

def agi_answer():
        print "ANSWER"

def agi_exec(app, args):
        print "EXEC %s \"%s\"" %(agi_escape_string(app), agi_escape_string(args))

File: web-interface/tpl/www/bloc/service/ipbx/asterisk/call_management/voicemenu/ipbxapplication/agi.php
<?php

#
# XiVO Web-Interface
# Copyright (C) 2006-2009  Proformatique <technique@proformatique.com>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#

?>
<div id="fd-ipbxapplication-agi" class="b-nodisplay">
<?php

$form = &$this->get_module('form');
$apparg_agi = $this->get_var('apparg_agi');

echo	$form->text(array('desc'	=> $this->bbf('fm_ipbxapplication_agi-command'),
			  'name'	=> 'ipbxapplication[agi][command]',
			  'labelid'	=> 'ipbxapplication-agi-command',
			  'size'	=> 15,
			  'default'	=> $apparg_agi['command']['default'])),

	$form->text(array('desc'	=> $this->bbf('fm_ipbxapplication_agi-args'),
			  'name'	=> 'ipbxapplication[agi][args]',
			  'labelid'	=> 'ipbxapplication-agi-args',
			  'size'	=> 15,
			  'default'	=> $apparg_agi['args']['default'])),

	$form->button(array('name'	=> 'add-ipbxapplication-agi',
			    'id'	=> 'it-add-ipbxapplication-agi',
			    'value'	=> $this->bbf('fm_bt-add')),
		      'onclick="xivo_ast_application_agi();"');

?>
</div>
